service/controller/chart/resource/release/create.go

```diff
@@ -28,126 +28,129 @@ func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange inte
-			err = r.helmClient.InstallReleaseFromTarball(ctx, tarballPath, ns, releaseState.Values, opts)
-			close(ch)
-		}()
-
-		select {
-		case <-ch:
-			// Fall through.
-		case <-time.After(3 * time.Second):
-			r.logger.LogCtx(ctx, "level", "debug", "message", "release still being created")
-			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
-			return nil
-		}
-		if helmclient.IsValidationFailedError(err) {
-			reason := err.Error()
-			reason = fmt.Sprintf("helm validation error: (%s)", reason)
-			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("helm release %#q failed, %s", releaseState.Name, reason))
-			addStatusToContext(cc, reason, validationFailedStatus)
-
+		// Release is failed so the status resource will check the Helm release.
+		if releaseContent.Status == helmclient.StatusFailed {
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("failed to create release %#q", releaseContent.Name))
 			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
 			resourcecanceledcontext.SetCanceled(ctx)
 			return nil
-		} else if helmclient.IsInvalidManifest(err) {
-			reason := err.Error()
-			reason = fmt.Sprintf("invalid manifest error: (%s)", reason)
-			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("helm release %#q failed, %s", releaseState.Name, reason))
-			addStatusToContext(cc, reason, invalidManifestStatus)
-
-			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
-			resourcecanceledcontext.SetCanceled(ctx)
-			return nil
-		} else if err != nil {
-			reason := err.Error()
-			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("helm release %#q failed", releaseState.Name), "stack", microerror.JSON(err))
-
-			releaseContent, err := r.helmClient.GetReleaseContent(ctx, ns, releaseState.Name)
-			if helmclient.IsReleaseNotFound(err) {
-				reason = fmt.Sprintf("helm error: (%s)", reason)
-				addStatusToContext(cc, reason, releaseNotInstalledStatus)
-
-				r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
-				resourcecanceledcontext.SetCanceled(ctx)
-				return nil
-			} else if err != nil {
-				return microerror.Mask(err)
-			}
-
-			// Release is failed so the status resource will check the Helm release.
-			if releaseContent.Status == helmclient.StatusFailed {
-				r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("failed to create release %#q", releaseContent.Name))
-				r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
-				resourcecanceledcontext.SetCanceled(ctx)
-				return nil
-			}
-			return microerror.Mask(err)
 		}
+		return microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("created release %#q", releaseState.Name))
 
-		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("created release %#q", releaseState.Name))
+	// We set the checksum annotation so the update state calculation
```

Comment:
This is actual change, we only should update the annotation after we upgrade the release successfully.